packages/go_router/lib/src/route.dart

```diff
@@ -227,3 +145,55 @@ class GoRoute {
   ) =>
       const SizedBox.shrink();
 }
+
+/// A route that displays a UI shell around the matching child route.
+///
+/// The widget built by the matching child route becomes to the child parameter
+/// of the [builder].
+class ShellRoute extends RouteBase {
+  /// Constructs a [ShellRoute].
+  ShellRoute({
+    required String path,
+    required this.builder,
+    this.defaultRoute,
+    GoRouterRedirect redirect = RouteBase._emptyRedirect,
+    List<RouteBase> routes = const <RouteBase>[],
+  }) : super._(
+          path: path,
+          routes: routes,
+          redirect: redirect,
+        );
+
+  /// The widget builder for a shell route.
+  final ShellRouteBuilder builder;
+
+  /// The relative path to the child route to navigate to when this route is
+  /// displayed. This allows the default child route to be specified without
+  /// using redirection.
+  final String? defaultRoute;
+}
+
+/// A route that displays all descendent [StackedRoute]s within its visual
+/// boundary, typically the UI shell of a [ShellRoute].
+///
+/// This route places a nested [Navigator] in the widget tree, where any
+/// descendent [StackedRoute]s are placed onto this
+/// Navigator instead of the root Navigator, which allows you to display a UI
+/// shell around a nested stack of routes if this route is a child route of
+/// [ShellRoute].
```

Comment:
Once I have a `NestedStackRoute`, is it possible to "break out" of its Navigator and add routes to the root Navigator? Should there be a comment about this?

Reply:
No, any descendent StackedRoutes will be placed onto this Navigator.

Reply:
Does that mean that once I have `NestedStackRoute` I can't have a screen within that hierarchy that covers the entire app? (Apologies for the noob question 😅)

Reply:
Only for child routes in your configuration, for example, the 

```
ShellRoute(path: '/home')
  -> StackedRoute(path: 'users')
    -> StackedRoute(path: 'user/:userId')
  -> NestedStackRoute(path: 'settings')
    -> StackedRoute(path: 'notifications')
```

In this example, the 'user/:userId' route is displayed on top of the root Navigator, but the '/settings/notifications' route is displayed within the subsection of the screen that 'settings' takes up. The '/home' route might build bottom navigation bar and want both options.

Reply:
Marking this as resolved since we are removing NestedStackRoute and StackedRoute